Technical/Logging/The-Tazama-Logging-Framework.md

```diff
@@ -1 +1,405 @@
+
+This microservice has an additional responsibility of batching the received logs before transmission. This has a wide range of benefits as ultimately, I/O operations are reduced which will generally have implications on from improved performance coming from less network calls which can also affect costs in cloud environments.
+
+### Implementation
+The [event-sidecar] and [lumberjack] are optional components in implementation. Perhaps hardware is limited or there may be some situations whereby running more microservices is not ideal, Tazama will still have a functional logging implementation if the [event-sidecar] and [lumberjack] are excluded from deployment. As mentioned before, you can opt in to using the [event-sidecar] and [lumberjack] for logging by specifying the sidecar host when initialising the logger service.
+
+#### Configuration
+
+##### With Event-Sidecar and Lumberjack
+
+###### Lumberjack
+
+The following services are required:
+
+- [NATS] - requires a [NATS] server to be running
+- [Elastic] - log destination
+
+A sample [.env](https://github.com/frmscoe/lumberjack/blob/ce6cda49bade1a0287bc0c603330e5fbb8159455/.env.example) is provided.
+
+| Variable Name         | Purpose                               | Example             |
+|-----------------------|---------------------------------------|---------------------|
+| NATS_SERVER           | Specifies the NATS server address      | `nats://localhost:4222` |
+| NATS_SUBJECT          | Defines the NATS subject/topic name    | `Lumberjack`        |
+| ELASTIC_SEARCH_VERSION| Specifies the Elasticsearch version   | `8.11`              |
+| ELASTIC_HOST          | Specifies the Elasticsearch host       | `http://localhost:9200` |
+| ELASTIC_USERNAME      | Username for Elasticsearch authentication | `elastic_user`   |
+| ELASTIC_PASSWORD      | Password for Elasticsearch authentication | `secretpassword` |
+| FLUSHBYTES            | Specifies the number of bytes to flush | `1024`              |
+
+
+###### Event-Sidecar
+A sample [.env](https://github.com/frmscoe/event-sidecar/blob/feb5e53b0801f60fa746e1720349cfb8c09e6c2b/.env.example) is provided.
+
+| Variable Name  | Purpose                              | Example                    |
+|----------------|--------------------------------------|----------------------------|
+| PORT           | Specifies the port number to use     | `8080`                     |
+| NATS_SERVER    | Specifies the NATS server address    | `nats://localhost:4222`    |
+| NATS_SUBJECT   | Defines the NATS subject/topic name  | `Lumberjack`               |
+
+> [!CAUTION]
+> Ensure the sidecar and lumberjack are pointing to the same instance of [NATS] and have the same subject
+
+###### Processor
```

Comment:
Perhaps expand the heading to "Implementing the logging in a processor" or something